api/v1alpha1/release_types.go

```diff
@@ -145,11 +148,19 @@ func (r *Release) HasSucceeded() bool {
 // IsDone returns a boolean indicating whether the Release's status indicates that it is done or not.
 func (r *Release) IsDone() bool {
 	condition := meta.FindStatusCondition(r.Status.Conditions, releaseConditionType)
-	if condition != nil {
-		return condition.Status != metav1.ConditionUnknown
+	return condition != nil && condition.Status != metav1.ConditionUnknown
+}
+
+// MarkDeployed sets the AllComponentsDeployed status in the Release to True or False with the provided reason and message.
+func (r *Release) MarkDeployed(status metav1.ConditionStatus, reason, message string) {
+	if status != metav1.ConditionUnknown {
+		r.setStatusConditionWithMessage(BindingDeploymentStatusConditionType, status, ReleaseReason(reason), message)
 	}
+}
 
-	return false
+// MarkDeploying sets the AllComponentsDeployed status in the Release to Unknown with the provided reason and message.
+func (r *Release) MarkDeploying(reason, message string) {
```

Comment:
I don't think we should allow to pass generic reasons in here.

Reply:
Why? I want to copy whatever reason gitops tells me

Reply:
What you really want is the message which is the one containing the number of deployments. The reason... I think we should have some more control over it but I'm ok with copying over. Let's reevaluate whenever gitops give us some code.

Reply:
No, that is why I pass message as well. I want both. But yes we can reevaluate later